Source/JavaScriptCore/jit/JITPropertyAccess.cpp

```diff
@@ -960,37 +961,46 @@ void JIT::emit_op_resolve_scope(const JSInstruction* currentInstruction)
                 loadPtr(Address(returnValueGPR, JSScope::offsetOfNext()), returnValueGPR);
         } else {
             ASSERT(localScopeDepth >= 8);
-            load32FromMetadata(bytecode, OpResolveScope::Metadata::offsetOfLocalScopeDepth(), scratch1GPR);
+            load32FromMetadata(bytecode, Metadata::offsetOfLocalScopeDepth(), scratch1GPR);
             auto loop = label();
             loadPtr(Address(returnValueGPR, JSScope::offsetOfNext()), returnValueGPR);
             branchSub32(NonZero, scratch1GPR, TrustedImm32(1), scratch1GPR).linkTo(loop, this);
         }
     } else {
         // Inlined fast path for common types.
-        uint32_t metadataOffset = m_profiledCodeBlock->metadataTable()->offsetInMetadataTable(bytecode);
-        addPtr(TrustedImm32(metadataOffset), GPRInfo::metadataTableRegister, metadataGPR);
+        static_assert(!(Metadata::offsetOfResolveType() % 4));
+        static_assert(!(Metadata::offsetOfGlobalLexicalBindingEpoch() % 4));
+        auto metadataAddress = computeGoodBaseAddressForMetadata<4>(bytecode, metadataGPR);
+
+        auto resolveTypeAddress = metadataAddress.withOffset(Metadata::offsetOfResolveType());
+        auto globalLexicalBindingEpochAddress = metadataAddress.withOffset(Metadata::offsetOfGlobalLexicalBindingEpoch());
```

Comment:
How do we know that addresses with these offsets are encodable?  The `computeGoodBaseAddressForMetadata` above did not take into account this additional `Metadata::offsetOfResolveType()` nor `Metadata::offsetOfGlobalLexicalBindingEpoch()`.  Should `computeGoodBaseAddressForMetadata` take an range / addition offset allowance parameter to cover these?

Reply:
In `computeGoodBaseAddressForMetadata`, the offset we check is encodable is `metadataOffset + sizeof(Metadata)` - so, assuming correct alignment of each field (which the `static_assert`s around the callsite check), any field within the metadata object should be encodable too.

Reply:
Oh, right.  I missed that.  You are right, `computeGoodBaseAddressForMetadata` already checks the worst case and guarantees that these offsets are reachable.